### PR TITLE
Made SavedState generic

### PR DESCRIPTION
**Pull request by hackaugusto**
This allows mypy to infer `ChainState`.

### CODE REVIEW

**Comment by LefterisJP** [comment]
I can doz reviews

File: raiden/storage/wal.py [hunk 1]
```diff
@@ -66,19 +59,20 @@ def restore_to_state_change(
 ST = TypeVar("ST", bound=State)
 
 
-class SavedState(NamedTuple):
+@dataclass(frozen=True)
+class SavedState(Generic[ST]):
```
Comment (LefterisJP) [inline raiden/storage/wal.py]: typo in the docstring not introduced by this PR but well you can fix it: `Saves the state the id of the state change` -> `Saves the state **and** the id of the state change`

File: raiden/storage/wal.py [hunk 2]
```diff
@@ -66,19 +59,20 @@ def restore_to_state_change(
 ST = TypeVar("ST", bound=State)
 
 
-class SavedState(NamedTuple):
+@dataclass(frozen=True)
```
Comment (LefterisJP) [inline raiden/storage/wal.py]: Can the `Generic[ST]` also be added to a `NamedTuple`? I generally don't know what's the point of changing something that has no functions to a frozen data class when a `NamedTuple` is enough.

Comment (hackaugusto) [reply raiden/storage/wal.py]: > Can the Generic[ST] also be added to a NamedTuple?

No, it will give an error.

**Pull request merged**